app/src/main/java/net/gsantner/markor/ui/SearchOrCustomTextDialogCreator.java

```diff
@@ -125,8 +125,15 @@ public static void showInsertTableRowDialog(final Activity activity, final boole
 
     public static void showSearchFilesDialog(Activity activity, File searchDir, Callback.a2<String, Integer> callback) {
         if (!SearchEngine.isSearchExecuting) {
+            final char[] password;
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
+                password = new AppSettings(activity).getDefaultPassword();
+            } else {
+                password = new char[0];
```

Comment:
Maybe `null` would be better flag?

Reply:
can be removed complete, for the default case there is default value at decl

Reply:
Remove completly